ntoskrnl/mm/section.c

```diff
@@ -3029,9 +3035,12 @@ MmCreateDataFileSection(PROS_SECTION_OBJECT *SectionObject,
             Segment->RawLength.QuadPart = MaximumSize.QuadPart;
             Segment->Length.QuadPart = PAGE_ROUND_UP(Segment->RawLength.QuadPart);
         }
+
+        /* We let the segment reference the file object */
+        ObDereferenceObject(FileObject);
+        FileObject = Segment->FileObject;
```

Comment:
When are these going to differ? If somebody creates a section for a file that already has one, but using a different file handle?

Reply:
When the FS creates a stream file object, you may have different file objects for the same on-disk file.